Continuous deployment to Heroku results in an error

##### **Overview of the issue**

I created a JHipster 7 demo project and added GitHub Actions CI with `jhipster ci-cd`. This worked great. Later, I re-ran `jhipster ci-cd` to add continuous delivery with Heroku. 

This resulted in a failure to deploy. Here's the PR to add it.

https://github.com/mraible/jhipster7-demo/pull/2

The build log kinda shows a failure:

https://github.com/mraible/jhipster7-demo/runs/2449382068?check_suite_focus=true#step:9:402

However, it was an email from Heroku that helped me find the real error.

<img width="585" alt="Screen Shot 2021-04-27 at 10 46 13 AM" src="https://user-images.githubusercontent.com/17892/116280357-cedb6980-a745-11eb-8bb3-11be1aae0c3f.png">

The error:

```
$ heroku releases:output v8
cp: cannot stat 'src/main/resources/config': No such file or directory
```

##### **Motivation for or Use Case**

CD should work with Heroku.

##### **Reproduce the error**

Add ci-cd to a new app and deploy it to Heroku.

##### **Suggest a Fix**

/cc @jkutner 

##### **JHipster Version(s)**

```
blog@0.0.1-SNAPSHOT /Users/mraible/dev/jhipster7-demo
└── generator-jhipster@7.0.1

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "monolith",
    "baseName": "blog",
    "jhipsterVersion": "7.0.1",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": ["cypress"],
    "pages": [],
    "creationTimestamp": 1619451584671,
    "serviceDiscoveryType": false,
    "reactive": false,
    "authenticationType": "jwt",
    "packageName": "org.jhipster.blog",
    "serverPort": "8080",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "buildTool": "maven",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "angularX",
    "withAdminUi": true,
    "clientTheme": "none",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "packageFolder": "org/jhipster/blog",
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "languages": ["en", "es"],
    "entities": ["Blog", "Post", "Tag"],
    "lastLiquibaseTimestamp": 1619453187000,
    "herokuAppName": "jhipster-7-demo",
    "herokuDeployType": "git",
    "herokuJavaVersion": "11"
  }
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}
entity Post {
  title String required
  content TextBlob required
  date Instant required
}
entity Tag {
  name String required minlength(2)
}
relationship ManyToOne {
  Blog{user(login)} to User
  Post{blog(name)} to Blog
}
relationship ManyToMany {
  Post{tag(name)} to Tag{post}
}

paginate Post, Tag with infinite-scroll

</pre>
</details>


##### **Environment and Tools**

openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)

git version 2.29.2

node: v14.16.1

npm: 6.14.12

Docker version 20.10.5, build 55c4c88

docker-compose version 1.28.5, build c4eb3a1f

I just got an email from Heroku saying a release phase failed. 

> A deployment for jhipster-7-demo failed due to a release phase command in release v6. To inspect the failure, check your release phase log in the dashboard or run 'heroku releases:output v6' in the CLI.
> 
> If you wish to retry the release, you can use the release retry CLI plugin.

It's strange that I just got this email when my last commit to my jhipster7-demo repo was two days ago. Maybe it was stuck in an outbox? Anyway, here's the release phase log.

```
Picked up JAVA_TOOL_OPTIONS: -XX:+UseContainerSupport -Xmx300m -Xss512k -XX:CICompilerCount=2 -Dfile.encoding=UTF-8 
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------< org.jhipster.blog:blog >-----------------------
[INFO] Building Blog 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- liquibase-maven-plugin:4.3.2:update (default-cli) @ blog ---
[INFO] ------------------------------------------------------------------------
[INFO] Loading artifacts into URLClassLoader
[INFO]   artifact: file:/app/.m2/repository/tech/jhipster/jhipster-framework/7.0.1/jhipster-framework-7.0.1.jar
[INFO]   <snip/>
[INFO]   artifact: file:/app/target/classes
[INFO]   artifact: file:/app/target/test-classes
[INFO] ------------------------------------------------------------------------
[project, pluginDescriptor]
[INFO] 
[INFO] 
[INFO] Liquibase Community 4.3.2 by Datical
[INFO] ####################################################
##   _     _             _ _                      ##
##  | |   (_)           (_) |                     ##
##  | |    _  __ _ _   _ _| |__   __ _ ___  ___   ##
##  | |   | |/ _` | | | | | '_ \ / _` / __|/ _ \  ##
##  | |___| | (_| | |_| | | |_) | (_| \__ \  __/  ##
##  \_____/_|\__, |\__,_|_|_.__/ \__,_|___/\___|  ##
##              | |                               ##
##              |_|                               ##
##                                                ## 
##  Get documentation at docs.liquibase.com       ##
##  Get certified courses at learn.liquibase.com  ## 
##  Free schema change activity reports at        ##
##      https://hub.liquibase.com                 ##
##                                                ##
####################################################
Starting Liquibase at 18:00:34 (version 4.3.2 #44 built at 2021-03-23 14:01+0000)
[INFO] Settings
----------------------------
[INFO]     driver: null
[INFO]     url: jdbc:postgresql://ec2-107-20-153-39.compute-1.amazonaws.com:5432/d15qu0crv220hs?password=fd86dffd6097a31c15cef8434d0f02a89c5ff79420e4ed389b5e998b483318d1&sslmode=require&user=vnigjftxbsvetu
[INFO]     username: vnigjftxbsvetu
[INFO]     password: *****
[INFO]     use empty password: false
[INFO]     properties file: null
[INFO]     properties file will override? false
[INFO]     prompt on non-local database? false
[INFO]     clear checksums? false
[INFO]     changeLogDirectory: null
[INFO]     changeLogFile: src/main/resources/config/liquibase/master.xml
[INFO]     context(s): null
[INFO]     label(s): null
[INFO]     number of changes to apply: 0
[INFO]     drop first? false
[INFO] ------------------------------------------------------------------------
[INFO] Executing on Database: jdbc:postgresql://ec2-107-20-153-39.compute-1.amazonaws.com:5432/d15qu0crv220hs?password=fd86dffd6097a31c15cef8434d0f02a89c5ff79420e4ed389b5e998b483318d1&sslmode=require&user=vnigjftxbsvetu
[INFO] Successfully acquired change log lock
[INFO] Configured classpath location file:/app/target/classes does not exist
[INFO] Configured classpath location file:/app/target/classes does not exist
[INFO] Configured classpath location file:/app/target/test-classes does not exist
[INFO] Configured classpath location file:/app/target/classes does not exist
[INFO] Reading from databasechangelog
[INFO] Successfully released change log lock
[INFO] Successfully acquired change log lock
Skipping auto-registration
[WARNING] Skipping auto-registration
[INFO] Table jhi_date_time_wrapper created
[INFO] ChangeSet config/liquibase/changelog/00000000000000_initial_schema.xml::00000000000002::jhipster ran successfully in 55ms
[INFO] Successfully released change log lock
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  24.374 s
[INFO] Finished at: 2021-05-01T18:00:40Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.liquibase:liquibase-maven-plugin:4.3.2:update (default-cli) on project blog: 
[ERROR] Error setting up or running Liquibase:
[ERROR] liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/20210426160625_added_entity_Blog.xml::20210426160625-1::jhipster:
[ERROR]      Reason: liquibase.exception.DatabaseException: ERROR: relation "blog" already exists [Failed SQL: (0) CREATE TABLE blog (id BIGINT NOT NULL, name VARCHAR(255) NOT NULL, handle VARCHAR(255) NOT NULL, user_id BIGINT, CONSTRAINT BLOG_PKEY PRIMARY KEY (id))]
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/generators/heroku/templates/Procfile.ejs
+++ b/generators/heroku/templates/Procfile.ejs
@@ -16,7 +16,4 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -%>
-web: java $JAVA_OPTS <% if (applicationType === 'gateway' || dynoSize === 'Free') { %>-Xmx256m<% } %> -jar <% if (buildTool === 'maven') { %>target<% } %><% if (buildTool === 'gradle') { %>build/libs<% } %>/*.jar --spring.profiles.active=prod,heroku<% if (buildTool == 'maven' && herokuDeployType == 'git') { %>,no-liquibase<% } %> <% if (prodDatabaseType === 'mongodb') { %>--spring.data.mongodb.database=$(echo "$MONGODB_URI" | sed "s/^.*:[0-9]*\///g")<% } %>
-<%_ if (buildTool == 'maven' && herokuDeployType == 'git' && (prodDatabaseType === 'postgresql' || prodDatabaseType === 'mysql' || prodDatabaseType === 'mariadb')) { _%>
-release: cp -R src/main/resources/config config && ./mvnw -ntp liquibase:update -Pprod,heroku
-<%_ } _%>
+web: java $JAVA_OPTS <% if (applicationType === 'gateway' || dynoSize === 'Free') { %>-Xmx256m<% } %> -jar <% if (buildTool === 'maven') { %>target<% } %><% if (buildTool === 'gradle') { %>build/libs<% } %>/*.jar --spring.profiles.active=prod,heroku <% if (prodDatabaseType === 'mongodb') { %>--spring.data.mongodb.database=$(echo "$MONGODB_URI" | sed "s/^.*:[0-9]*\///g")<% } %>

